reference-implementation/lib/writable-stream.js

```diff
@@ -621,6 +596,15 @@ function WritableStreamDefaultWriterCloseWithErrorPropagation(writer) {
   return WritableStreamDefaultWriterClose(writer);
 }
 
+function WritableStreamDefaultWriterEnsureReadyPromiseRejected(writer, error) {
+  if (writer._readyPromiseState === 'pending') {
```

Comment:
Yeah, I understand. It is indeed a good goal to be able to maintain the state out of band, but it just got really complex and error-prone. At least I personally was not able to figure it out.